src/cljs/main/broadfcui/common.cljs

```diff
@@ -225,53 +225,81 @@
                :title tooltip}
         text]))})
 
-(react/defc FoundationInfoBox
-  {:component-will-mount
+(react/defc FoundationIconDropdown
+  {:close
    (fn [{:keys [locals]}]
-     (swap! locals assoc :infobox-id (gensym "infobox-")))
+     (.foundation (js/$ (:dropdown-element @locals)) "close"))
+   :get-default-props
+   (fn []
+     {:icon-name :information})
+   :component-will-mount
+   (fn [{:keys [locals]}]
+     (swap! locals assoc :dropdown-id (gensym "dropdown-")))
+   :render
+   (fn [{:keys [props locals]}]
+     [:button {:className "button-reset" :data-toggle (:dropdown-id @locals)
+               :style {:cursor "pointer" :padding "0 0.5rem"
+                       :fontSize "16px" :lineHeight "1rem"}}
+      (icons/icon {:style {:color (:icon-color props)}} (:icon-name props))])
    :component-did-mount
-   (fn [{:keys [locals this props]}]
-     (let [infobox-container (.createElement js/document "div")]
-       (swap! locals assoc :infobox-container infobox-container)
-       (.insertBefore js/document.body infobox-container (utils/get-app-root-element))
+   (fn [{:keys [this locals]}]
+     (let [dropdown-container (.createElement js/document "div")]
+       (swap! locals assoc :dropdown-container dropdown-container)
+       (.insertBefore js/document.body dropdown-container (utils/get-app-root-element))
+       (this :-render-dropdown)
+       (.foundation (js/$ (:dropdown-element @locals)))))
+   :component-will-receive-props
+   (fn [{:keys [this locals]}]
+     (this :-render-dropdown))
+   :component-will-unmount
+   (fn [{:keys [locals]}]
+     (react/unmount-component-at-node (:dropdown-container @locals))
+     (.remove (:dropdown-container @locals)))
+   :-render-dropdown
+   (fn [{:keys [this props state after-update locals]}]
+     (let [{:keys [contents]} props
+           {:keys [dropdown-container dropdown-id]} @locals]
        (react/render
         (react/create-element
          ;; empty string makes react attach a property with no value
-         [:div {:className "dropdown-pane" :id (:infobox-id @locals) :data-dropdown ""
+         [:div {:className "dropdown-pane" :id dropdown-id :data-dropdown ""
                 :ref (utils/create-element-ref-handler
                       {:store locals
                        :key :dropdown-element
                        :did-mount
```

Comment:
Would it be easier to read if we made this a named method and just called it here, since it's long and skinny?